src/SipHash.chpl

```diff
@@ -69,20 +81,32 @@ module SipHash {
     return c;
   }
   
-  proc sipHash64(msg: [] uint(8), D): uint(64) {
+  proc sipHash64(msg: [] ?t, D): uint(64) {
     var (res,_) = computeSipHashLocalized(msg, D, 8);
     return res;
   }
 
-  proc sipHash128(msg: [] uint(8), D): 2*uint(64) {
+  proc sipHash128(msg: [] ?t, D): 2*uint(64) {
     return computeSipHashLocalized(msg, D, 16);
   }
 
-  private proc computeSipHashLocalized(msg: [] uint(8), D, param outlen: int) {
+  /*
+   * Compute hash of a single value
+   */
+  proc sipHash64(in val): uint(64) {
+    var (res,_) = computeSipHash(c_ptrTo(val), 0..#1, 8, 8);
+    return res;
+  }
+  
+  proc sipHash128(in val): 2*uint(64) {
+    return computeSipHash(c_ptrTo(val), 0..#1, 16, 8);
+  }
+  
+  private proc computeSipHashLocalized(msg: [] ?t, D, param outlen: int) {
     if contiguousIndices(msg) {
       ref start = msg[D.low];
       if D.high < D.low {
-        return computeSipHash(c_ptrTo(start), 0..#0, outlen);
+        return computeSipHash(c_ptrTo(start), 0..#0, outlen, numBytes(t));
```

Comment:
Note that `numBytes` does not work for `bool` today since the size is implementation defined. I'm not sure if this is a problem for you today since I'm not sure if you're hashing bool arrays.

Reply:
Oh, good to know. Right, we're not hashing bool arrays currently, but we'll have to watch out for this.